### PR TITLE
lib is not prepended on windows automatically, so to dynamically link…

### LINKED ISSUE #96: Dynamic linking doesn't work on windows
Your code expects a z3.lib and z3.dll:

https://github.com/prove-rs/z3.rs/blob/3c6ab61482c64b083e79e6ccf550a5fb1e87b0a9/z3-sys/src/lib.rs#L1562-L1563

But building z3 outputs a libz3.lib and libz3.dll.

I checked how debian [packs libz3](https://packages.debian.org/de/sid/libz3-dev), and they ship a libz3.so, not a z3.so. Why do you try to link against z3 instead of libz3, and on what platforms does that work?

### PR DESCRIPTION
**Pull request by Trolldemorted**
… a z3 install we must have an explicit libz3 link

Should fix #96, I tested it locally with a dynamically linked, self built z3

### CODE REVIEW

**Comment by Trolldemorted** [comment]
Now it works on windows both statically and dynamically

File: z3-sys/build.rs [hunk 1]
```diff
@@ -53,18 +53,10 @@ fn build_z3() {
...
-        let from = lib.join("libz3.lib");
-        let to = lib.join("z3.lib");
-        std::fs::rename(&from, &to).expect(&format!(
-            "failed to rename `{}` to `{}`",
-            from.display(),
-            to.display()
-        ));
+        println!("cargo:rustc-link-lib=static=libz3");
+    } else {
+        println!("cargo:rustc-link-lib=static=z3");
```
Comment (fitzgen) [inline z3-sys/build.rs:60]: Nice clean up, thanks.

File: z3-sys/src/lib.rs [hunk 2]
```diff
@@ -7482,3 +7481,11 @@ extern "C" {
...
     pub fn Z3_qe_lite(c: Z3_context, vars: Z3_ast_vector, body: Z3_ast) -> Z3_ast;
 }
+
+#[cfg(not(windows))]
+#[link(name = "z3")]
+extern "C" {}
+
+#[cfg(windows)]
+#[link(name = "libz3")]
+extern "C" {}
```
Comment (fitzgen) [inline z3-sys/src/lib.rs:7491]: Are these still required even though we are printing the cargo linking commands? I would assume that they are not.

**Review by fitzgen — COMMENTED** [review]
Thanks! One question below before we merge this.

File: z3-sys/src/lib.rs (hunk 2 above)
Comment (Trolldemorted) [reply z3-sys/src/lib.rs:7491]: Is that properly done for dynamic links too? The function where I changed that is guarded by `#[cfg(feature = "static-link-z3")]`

Comment (fitzgen) [reply z3-sys/src/lib.rs:7491]: Ah, good catch. You're correct.

**Pull request merged**